### PR TITLE
GH-1371: Lint more.

### PR DESCRIPTION
**Pull request by tbl3rd**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-1371

### Changes

- Added an alias for the eastwood linter.
- Ran kibit and eastwood and kondo and acquiesced to most of their demands.
- Tweaked a couple of other things noticed along the way. 
- Changed `lint` target to `format` and run it during builds.
- New `lint` target runs all 3 linters with their configuration files.

### Review Instructions

- The linters still complain.
- Please run them and let me know what you think.
- We can probably suppress some of the remaining complaints.
- We may not want to keep all of these changes.

### CODE REVIEW

File: api/src/wfl/module/copyfile.clj [hunk 1]
```diff
@@ -101,7 +101,7 @@
 (defn create-copyfile-workload!
   "Use transaction TX to add the workload described by REQUEST."
   [tx {:keys [items common] :as request}]
-  (letfn [(nil-if-empty [x] (if (empty? x) nil x))
+  (letfn [(nil-if-empty [x] (when (seq x) x))
```
Comment (ehigham) [inline api/src/wfl/module/copyfile.clj:104]: that's just `not-empty`

File: api/test/wfl/unit/workloads_test.clj [hunk 2]
```diff
@@ -5,13 +5,13 @@
...
 
 (deftest test-saved-before?
-  "in version 0.4.0, wgs workloads were serialized using CromwellWorkload table"
-  (is (not (workloads/saved-before? "0.4.0" {:version "0.4.0"})))
-  (is (not (workloads/saved-before? "0.4.0" {:version "1.0.0"})))
-  (is (not (workloads/saved-before? "0.4.0" {:version "0.4.1"})))
-  (is (workloads/saved-before? "0.4.0" {:version "0.3.8"}))
-  (is (workloads/saved-before? "0.4.0" {:version "0.0.0"}))
-  (is (thrown? ExceptionInfo (workloads/saved-before? "0.4" {:version "0.0.0"}))))
+  (testing "in version 0.4.0, wgs workloads were serialized using CromwellWorkload table"
```
Comment (ehigham) [inline api/test/wfl/unit/workloads_test.clj:8]: I really don't like the "testing blah blah blah" in tests like this. It adds nothing. Rename the test if you think it's not clear.
Also, this change was not limited to WGS

**Review by ehigham — APPROVED** [review]
OK i guess? Not sure how much value some of these changes add.

**Comment by ehigham** [comment]
If the linters complain perhaps you could fix that and have them run on the lint target?

File: api/src/wfl/module/copyfile.clj (hunk 1 above)
Comment (okotsopoulos) [reply api/src/wfl/module/copyfile.clj:104]: Clojure says that `(seq x)` is the preferred idiom for testing a non-empty collection: https://clojuredocs.org/clojure.core/seq#example-542692d5c026201cdc327079

Comment (ehigham) [reply api/src/wfl/module/copyfile.clj:104]: Good to know.

**Review by rhiananthony — APPROVED** [review]
👍🏼

File: api/src/wfl/module/copyfile.clj (hunk 1 above)
Comment (tbl3rd) [reply api/src/wfl/module/copyfile.clj:104]: I like `not-empty` for this.

File: api/test/wfl/unit/workloads_test.clj (hunk 2 above)
Comment (tbl3rd) [reply api/test/wfl/unit/workloads_test.clj:8]: I just wrapped what looked like an attempt at a docstring.
My goal was to reduce reports from the linter without changing behavior.
Except where a linter discovered a problem: then I tried to fix the problem.

File: api/src/wfl/module/copyfile.clj (hunk 1 above)
Comment (tbl3rd) [reply api/src/wfl/module/copyfile.clj]: FYI: `seq` returns a sequence on the collection, but `not-empty` returns the original collection.

File: api/src/wfl/module/covid.clj [hunk 3]
```diff
@@ -25,20 +25,20 @@
 ;; interfaces
```
Comment (okotsopoulos) [inline api/src/wfl/module/covid.clj:25]: Many / most of these multimethods moved FYI:

https://github.com/broadinstitute/wfl/blob/develop/api/src/wfl/stage.clj

File: api/src/wfl/module/covid.clj [hunk 4]
```diff
@@ -50,59 +50,59 @@
...
+  "Update the `sink` with the `executor`."
+  (fn [_executor sink] (:type sink)))
 
 (defmulti load-sink!
-  "Use `tx` to load the workload sink with `sink_type`."
-  (fn [tx workload] (:sink_type workload)))
+  "Use `tx` to load the `workload` sink with `sink_type`."
+  (fn [_tx workload] (:sink_type workload)))
 
 ;; Processing "stage" operations
```
Comment (okotsopoulos) [inline api/src/wfl/module/covid.clj:103]: This one too:

https://github.com/broadinstitute/wfl/blob/develop/api/src/wfl/stage.clj

File: api/src/wfl/module/covid.clj (hunk 3 above)
Comment (tbl3rd) [reply api/src/wfl/module/covid.clj:25]: Don't worry about it.  I'll just lint again before merging.

**Review by rexwangcc — APPROVED** [review]

File: api/test/wfl/tools/workflows.clj [hunk 5]
```diff
@@ -42,8 +42,8 @@
    type object))
 
 (defn foldl
-  "Use the workflow `type` to left-fold the `object`, calling `f` on the values
-   with primitive types with the current state."
+  "Use ternary function `f` to reduce `object` tree of `type` starting
+  from `init` while preserving the types of intermediate reductions."
```
Comment (tbl3rd) [inline api/test/wfl/tools/workflows.clj:46]: @ehigham (-How wrong is this new docstring?-)

Comment (ehigham) [reply api/test/wfl/tools/workflows.clj:46]: While preserving the types of intermediate reductions? what does that mean?

Comment (ehigham) [reply api/test/wfl/tools/workflows.clj:46]: You could change `f` to take [result-so-far [type value]]` so it looks like "reduce" if that helps.

**Review by ehigham — APPROVED** [review]

File: api/test/wfl/tools/workflows.clj (hunk 5 above)
Comment (tbl3rd) [reply api/test/wfl/tools/workflows.clj:46]: The string tags are the "types".
Reductions are the intermediate result subtrees such as might be returned by `clojure.core/reductions`.
This differs from a generic left fold (or `reduce`) in that it treats one of the parameters differently.
Perhaps in your terminology: this is a left fold over a tree with an implicit lens.
(FWIW: I don't care where the grouping happens. This doesn't have to look like `reduce`.)

Comment (tbl3rd) [reply api/test/wfl/tools/workflows.clj:46]: Mostly, I just want some mention of the roles of the parameters to `foldl`,
and to cast it in Clojure terms so it is familiar to the team (or me, at least).
Please give me a thumbs up, a docstring more acceptable to you, or a "revert it".
Thanks.

**Pull request merged**